src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -1,652 +1,399 @@
+      if (!res && pathFormControl.value && !this.stripACLWarningSent) {
+        this.ws.call('filesystem.acl_is_trivial', [pathFormControl.value])
+          .pipe(untilDestroyed(this)).subscribe((res) => {
+            if (!res) {
+              this.stripACLWarningSent = true;
+              this.showStripAclWarning();
+            }
+          });
       }
     });
+  }
 
-    setTimeout(() => {
-      if (entityForm.formGroup.controls['timemachine'].value) {
-        this.isTimeMachineOn = true;
-      }
-    }, 700);
+  showStripAclWarning(): void {
+    this.dialog.confirm({
+      title: helptextSharingSmb.stripACLDialog.title,
+      message: helptextSharingSmb.stripACLDialog.message,
+      hideCheckBox: true,
+      buttonMsg: helptextSharingSmb.stripACLDialog.button,
+      hideCancel: true,
+    }).pipe(untilDestroyed(this)).subscribe();
+  }
 
-    entityForm.formGroup.controls['purpose'].valueChanges.pipe(untilDestroyed(this)).subscribe((res) => {
-      this.clearPresets();
-      for (const item in this.presets[res].params) {
-        this.presetFields.push(item as keyof SmbShare);
-        const ctrl = entityForm.formGroup.controls[item];
-        if (ctrl && item !== 'auxsmbconf') {
-          ctrl.setValue(this.presets[res].params[item as keyof SmbShare]);
-          ctrl.disable();
-        }
+  clearPresets(): void {
+    for (const item of this.presetFields) {
+      this.form.get(item).enable();
+    }
+    this.presetFields = [];
+  }
+
+  setSmbShareForEdit(smbShare: SmbShare): void {
+    this.existingSmbShare = smbShare;
+    this.hostsAllowOnLoad = smbShare.hostsallow ? [...smbShare.hostsallow] : [];
+    this.mangle = smbShare.aapl_name_mangling;
+    this.hostsDenyOnLoad = smbShare.hostsdeny ? [...smbShare.hostsdeny] : [];
+    this.isTimeMachineOn = smbShare.timemachine;
+    this.title = helptextSharingSmb.formTitleEdit;
+    const index = this.namesInUse.findIndex((name) => name === smbShare.name);
+    if (index >= 0) {
+      this.namesInUse.splice(index, 1);
+    }
+    this.form.get('aapl_name_mangling').valueChanges.pipe(untilDestroyed(this)).subscribe((value) => {
+      if (value !== this.mangle && !this.mangleWarningSent) {
+        this.mangleWarningSent = true;
+        this.dialog.confirm({
+          title: helptextSharingSmb.manglingDialog.title,
+          message: helptextSharingSmb.manglingDialog.message,
+          hideCheckBox: true,
+          buttonMsg: helptextSharingSmb.manglingDialog.action,
+          hideCancel: true,
+        }).pipe(untilDestroyed(this)).subscribe();
       }
     });
+    this.form.patchValue(smbShare);
   }
 
-  afpConfirm(): void {
-    const afpControl = this.entityForm.formGroup.controls['afp'];
+  /* If user blurs name field with empty value, try to auto-populate based on path */
+  setEmptyNameFromPath(): void {
+    const nameControl = this.form.get('name');
```

Comment:
Listener similar to what you have in `ngOnInit`? Can be combined?

Reply:
Since, we're changing the value of the `name` control, setting up a subscription to valuechanges triggers recursion issues here and inside `ix-input`

Reply:
When you patchValue in form there is an optional argument that you can set no to trigger updates. emitEvent: false